Main.hs

```diff
@@ -11,28 +11,31 @@ import qualified Data.List as L
 import Text.Printf (printf)
 
 multByIndex :: [Int] -> [Int]
-multByIndex = undefined
+multByIndex a = zipWith (*) [0..] a  
 
 powerByIndex :: [Int] -> [Int]
-powerByIndex = undefined
-
+powerByIndex a = zipWith (^) a [0..] 
 productOfDifference :: [Int] -> Int
-productOfDifference = undefined
+productOfDifference a
+    | length(a) < 2 = error "too short"
+    | otherwise = foldr (*) 1 (zipWith (-) (a) (tail a))
 
 isSorted :: [Int] -> Bool 
-isSorted xs = undefined
+isSorted a = and (zipWith (<=) a (tail a))
 
 countElement :: Int -> [Int] -> Int 
-countElement = undefined
+countElement a x = length(filter (check a) x)
+    where check eq var = (eq == var)
```

Comment:
actually you can use prefix notation of equality operator as a parameter for filter, without introducing check function

a == b ~ ~ `==` a b

Reply:
try to redo this task

Reply:
> try to redo this task

Changed the code.